pkg/ingester/client/circuitbreaker.go

```diff
@@ -38,41 +39,41 @@ var (
 	}
 )
 
-func NewCircuitBreaker(addr string, cfg CircuitBreakerConfig, metrics *Metrics, logger log.Logger) grpc.UnaryClientInterceptor {
+func NewCircuitBreaker(inst ring.InstanceDesc, cfg CircuitBreakerConfig, metrics *Metrics, logger log.Logger) grpc.UnaryClientInterceptor {
+	// Initialize each of the known labels for circuit breaker metrics for this particular ingester
+	transitionOpen := metrics.circuitBreakerTransitions.WithLabelValues(inst.Id, circuitbreaker.OpenState.String())
```

Comment:
Shall we reset all metrics when creating new circuit breaker?

We may be creating new breaker for same instance ID as before (if client was closed in the meantime), but state of circuit breaker and metrics will not match in that case.

Reply:
This particular metric is a counter of transitions to a particular state, not a gauge of the current state.

Reply:
👍 Thanks, I missed that.